libbeat/processors/add_docker_metadata/add_docker_metadata.go

```diff
@@ -172,11 +171,15 @@ func (d *addDockerMetadata) Run(event *beat.Event) (*beat.Event, error) {
 		}
 
 		if len(container.Labels) > 0 {
-			labels := common.MapStr{}
+			//dedot first
+			outputLabels := common.MapStr{}
 			for k, v := range container.Labels {
-				safemapstr.Put(labels, k, v)
+				// This is necessary so that ES does not interpret '.' fields as new
+				// nested JSON objects, and also makes this compatible with ES 2.x.
+				label := common.DeDot(k)
```

Comment:
You can probably reuse the `DeDotLabels` helper in the docker module, move it to some common place in `libbeat`.

Reply:
Good point, should add_docker_metadata and docker container module uses the same `labels.dedot` config? Hmm I added a separate one into add_docker_metadata in libbeat.